zproject/urls.py

```diff
@@ -220,6 +218,12 @@
         {'PATCH': 'patch_bot_backend',
          'DELETE': 'deactivate_bot_backend'}),
 
+) + patterns('zerver.views.upload',
+    url(r'^user_uploads/(?P<realm_id_str>(\d*|unk))/(?P<filename>.*)', 'rest_dispatch',
+            {'GET': 'api_get_uploaded_file'}),
```

Comment:
The `rest_dispatch' wrapper checks for the presence of /ap` in the URL for authenticating it via HTTP BASIC. So we might have to add a special case for /user_uploads